velox/exec/fuzzer/AggregationFuzzerBase.cpp

```diff
@@ -318,11 +338,30 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputDataWithRowNumber(
           generator->generate(signature.args, vectorFuzzer_, rng_, pool_.get());
     }
 
-    for (auto i = children.size(); i < types.size() - 1; ++i) {
+    // Only iterate up to right before row_number column
+    auto typesSize = types.size() - 1;
+    // If k-range frame, only iterate up to
+    // right before row_number, startColumn,
+    // and endColumn
+    if (hasKRangeFrameBound) {
+      typesSize = types.size() - 3;
+    }
+
+    for (auto i = children.size(); i < typesSize; ++i) {
       children.push_back(vectorFuzzer_.fuzz(types[i], size));
     }
     children.push_back(vectorMaker.flatVector<int64_t>(
         size, [&](auto /*row*/) { return rowNumber++; }));
+
+    if (hasKRangeFrameBound) {
```

Comment:
The logic here is more complicated depending on ASC/DESC sort order and PRECEDING or FOLLOWING for the frame (Frame start could be FOLLOWING as well and frame end could be PRECEDING as well). 
https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/window/tests/WindowTestBase.cpp#L216

I feel we should delay the computation of these columns to where the range frame clause is generated.